Public/Src/Sandbox/Windows/DetoursServices/DetouredFunctions.cpp

```diff
@@ -1850,6 +1852,40 @@ BOOL WINAPI Detoured_CreateProcessW(
             lpProcessInformation);
     }
 
+    // If the process to be created is configured to breakaway from the current
+    // job object, we use the regular process creation, and set the breakaway flag
+    if (!g_processNamesToBreakAwayFromJob->empty())
+    {
+        std::wstring imageName = GetImageName(lpApplicationName, lpCommandLine);
+
+        // An empty string means we couldn't find a candidate, or the path was malformed.
+        // In this case we just let the regular detoured create process take control
+        if (imageName != L"")
+        {
+            std::unordered_set<wstring, CaseInsensitiveStringHasher, CaseInsensitiveStringComparer>::iterator result;
+            result = g_processNamesToBreakAwayFromJob->find(imageName);
+            
+            if (result != g_processNamesToBreakAwayFromJob->end())
+            {
+                Dbg(L"Allowing process to breakaway from job object. Image name: '%s'", imageName.c_str());
+

```

Comment:
Do you intentionally want to send this debugging message to the log during production?

Reply:
I'll enclose it in a SUPER_VERBOSE #if

---
In reply to: [339169069](https://github.com/microsoft/BuildXL/pull/1099#discussion_r339169069) [](ancestors = 339169069)